olimex/teres_i/default.nix

```diff
@@ -0,0 +1,30 @@
+{ config, lib, pkgs, ...}:
+
+{
+  hardware.deviceTree = {
+    name = lib.mkDefault "allwinner/sun50i-a64-teres-i.dts";
+    enable = lib.mkDefault true;
+  };
+
+  boot = {
+    consoleLogLevel = lib.mkDefault 7;
+
+    kernelPackages = lib.mkDefault pkgs.linuxPackages_latest;
+    kernelParams = lib.mkDefault ["console=ttyS0,115200n8"];
```

Comment:
This will make serial the main way of seeing boot logs (other ttys will only see a getty later on). Is this the preferred way to access it? Does the board have other means like HDMI output that might be used? I.e. if someone doesn't have a hdmi adapter.

Reply:
Also see https://github.com/NixOS/nixos-hardware/pull/745#discussion_r1341922235

Reply:
> This will make serial the main way of seeing boot logs (other ttys will only see a getty later on). Is this the preferred way to access it? Does the board have other means like HDMI output that might be used? I.e. if someone doesn't have a serial adapter. -- @Mic92 (https://github.com/NixOS/nixos-hardware/pull/750#discussion_r1348578693)

it has micro-hdmi and eDP both of which seem to show the full initialization of the OS.

Serial adapter is essential to be able to use the system as far as i know every teres user has one.

> Also see https://github.com/NixOS/nixos-hardware/pull/745#discussion_r1341922235 -- @Mic92 (https://github.com/NixOS/nixos-hardware/pull/750#discussion_r1348581466)

Afaik it's not relevant, the initialization of rockchip differs from sunxi.

Reply:
Allwinner and Rockchip are both irrelevant here. It's a kernel feature.

The reason this "works" is because this is a no-op, functionally speaking.

`lib.mkDefault` on a list that is being used by the modules system in any other way is extremely likely to mean the value is not being used. As `boot.kernelParams` is being used for many other reason, this line does nothing.